staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -447,6 +447,12 @@ func (s unstructuredJSONScheme) decodeToList(data []byte, list *UnstructuredList
 	delete(list.Object, "items")
 	list.Items = make([]Unstructured, 0, len(dList.Items))
 	for _, i := range dList.Items {
+		// make sure the data can decode into ObjectMeta before we return,
+		// so we don't silently truncate schema errors in metadata later with accesser get/set calls
+		v := &metadataOnlyObject{}
```

Comment:
Maybe moving this check to [metadata_decoder.go](https://github.com/liggitt/kubernetes/blob/master/staging/src/k8s.io/cli-runtime/pkg/resource/metadata_decoder.go) by decorating `decodeToList`.